File: 其他/constructArr/contructArr.c
int* constructArr(int* a, int aSize, int* returnSize){
    if(aSize < 2){
        return a;
    }
    returnSize = &aSize;
    int * result = (int*)malloc(sizeof(int)*aSize);
    int prefix[aSize];
    int suffix[aSize];
    prefix[0] = a[0];
    suffix[aSize-1] = a[aSize-1];
    for(int i=1;i<aSize;i++){
        prefix[i] = prefix[i-1] * a[i];
    }
    for(int j=aSize-2;j>0;j--){
        suffix[j] = suffix[j+1] * a[j];
    }
    result[0] = suffix[1];
    result[aSize-1] = prefix[aSize-2];
    for(int k=1;k<aSize-1;k++){
        result[k] = prefix[k-1] * suffix[k+1];
    }
    for(int i=0;i<*returnSize;i++){
        printf("%d\n", result[i]);
    }
    return result;
}

File: 其他/constructArr/contructArr.java
class Solution {
    public int[] constructArr(int[] a) {
        int len = a.length;
        if(len < 2){
            return a;
        }
        int[] prefix = new int[len];
        prefix[0] = a[0];
        int[] suffix = new int[len];
        int[] result = new int[len];
        suffix[len-1] = a[len-1];
        for(int idx=1;idx<len;idx++){
            prefix[idx] = prefix[idx-1] * a[idx];
        }
        for(int idx=len-2;idx>0;idx--){
            suffix[idx] = suffix[idx+1] * a[idx];
        }
        result[0] = suffix[1];
        result[len-1] = prefix[len-2];
        for(int idx=1;idx<len-1;idx++){
            result[idx] = prefix[idx-1] * suffix[idx+1];
        }
        return result;
    }
}